### PR TITLE
vectorstores: add redisvector implementation

### PR DESCRIPTION
**Pull request by CrazyWr**
### PR Checklist

- [x] Read the [Contributing documentation](https://github.com/tmc/langchaingo/blob/main/CONTRIBUTING.md).
- [x] Read the [Code of conduct documentation](https://github.com/tmc/langchaingo/blob/main/CODE_OF_CONDUCT.md).
- [x] Name your Pull Request title clearly, concisely, and prefixed with the name of the primarily affected package you changed according to [Good commit messages](https://go.dev/doc/contribute#commit_messages) (such as `memory: add interfaces for X, Y` or `util: add whizzbang helpers`).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `Fixes #123`).
- [x] Describes the source of new concepts.
- [x] References existing implementations as appropriate.
- [x] Contains test coverage for new functions.
- [x] Passes all [`golangci-lint`](https://golangci-lint.run/) checks.

### CODE REVIEW

**Review by devalexandre — COMMENTED** [review]
Could you create a example in example folder ?

**Comment by CrazyWr** [comment]
Resolves #581

File: examples/redis-vectorstore-example/go.mod [hunk 1]
```diff
@@ -0,0 +1,41 @@
+module github.com/tmc/langchaingo/examples/redis-vectorstore-example
+
+go 1.22.0
+
+toolchain go1.22.1
+
+// NOTE: remove the following line to use the official (rather than local development) version
+replace github.com/tmc/langchaingo => ../..
```
Comment (tmc) [inline examples/redis-vectorstore-example/go.mod:8]: you can use a go.work file to manage this locally. A doc suggesting a dev workflow would be super useful.

Comment (CrazyWr) [reply examples/redis-vectorstore-example/go.mod]: rm example temporarily，because of #782

**Review by tmc — APPROVED** [review]
Lovely implementation, LGTM. Please follow on and address your TODOs and please aim to comment all exported symbols.

**Comment by tmc** [comment]
Rebased and go mod tidy'd.

**Pull request merged**